api/v1alpha3/types.go

```diff
@@ -145,42 +140,48 @@ type IngressRule struct {
 // IngressRules is a slice of Azure ingress rules for security groups.
 type IngressRules []*IngressRule
 
-// PublicIP defines an Azure public IP address.
-type PublicIP struct {
-	ID        string `json:"id,omitempty"`
-	Name      string `json:"name,omitempty"`
-	IPAddress string `json:"ipAddress,omitempty"`
-	DNSName   string `json:"dnsName,omitempty"`
+// LoadBalancerSpec defines an Azure load balancer.
+type LoadBalancerSpec struct {
+	ID          string       `json:"id,omitempty"`
+	Name        string       `json:"name,omitempty"`
+	SKU         SKU          `json:"sku,omitempty"`
+	FrontendIPs []FrontendIP `json:"frontendIPs,omitempty"`
```

Comment:
That's a good question. From an Azure perspective, a load balancer can have more than 1 frontend IP. For the API Server LB, we only allow one frontend IP currently but my reasoning for making the type an array was that 1) we might need more than 1 in the future and 2) we might reuse the LoadBalancerSpec type for other LoadBalancer specs in the future so it's easier to make it an array from the start so we don't have to make API changes later. I agree it makes the UX a little awkward because an array implies you can specify more than one. What do you think?

Reply:
matching the azure types is a good reason, its fine to keep it an array as long as we have a good error message in the validation and document it somewhere (maybe a comment on the type).